### PR TITLE
provides tests for updating a submodel element in root

### PR DESCRIPTION
**Pull request by ahoimariew**
added test for root submodel element unpdates in
- SubmodelServiceSuite
- SubmodelRepositorySubmodelHTTPTestSuite

author:
@ahoimariew

### CODE REVIEW

File: basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java [hunk 1]
```diff
@@ -229,6 +229,22 @@ public void updateNonFileSME() throws FileNotFoundException, IOException, ParseE
...
+	    CloseableHttpResponse creationResponse = BaSyxSubmodelHttpTestUtils.createSubmodel(getURL(), submodelJSON);
+	    assertSubmodelCreationReponse(submodelJSON, creationResponse);
+	        
+	    String element = getJSONValueAsString("PropertySubmodelElementUpdate.json");
+	        
+	    String idShortPathPropertyInRootSme = "MaxRotationSpeed";
+	    CloseableHttpResponse updatedResponse = updateElement(createSpecificSubmodelElementURL(DummySubmodelFactory.SUBMODEL_TECHNICAL_DATA_ID,idShortPathPropertyInRootSme), element);
+	    assertEquals(HttpStatus.NO_CONTENT.value(), updatedResponse.getCode());
+	    CloseableHttpResponse fetchedResponse = BaSyxHttpTestUtils.executeGetOnURL(createSpecificSubmodelElementURL(DummySubmodelFactory.SUBMODEL_TECHNICAL_DATA_ID,idShortPathPropertyInRootSme));
+	    BaSyxHttpTestUtils.assertSameJSONContent(element, BaSyxHttpTestUtils.getResponseAsString(fetchedResponse));
```
[the remark this answers is not in the document]
Comment (ahoimariew) [reply basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java:246]: I did

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java [hunk 2]
```diff
@@ -362,6 +362,23 @@ public void updateNonFileSME() {
...
+    public void updateNonNestedSME() {
+        Submodel technicalSubmodel = DummySubmodelFactory.createTechnicalDataSubmodel();
+        SubmodelService submodelService = getSubmodelService(technicalSubmodel);
+        Property property = createDummyProperty("test123");
+
+        submodelService.createSubmodelElement(property);
+
+        String idShortPathRootSME = "test123";
+
+        Property newProperty = SubmodelServiceHelper.createDummyProperty("test123", "arbitraryValue", DataTypeDefXsd.STRING);
```
Comment (mdanish98) [inline basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java:375]: Please rename newProperty to expectedUpdatedProperty

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java [hunk 3]
```diff
@@ -362,6 +362,23 @@ public void updateNonFileSME() {
...
+        submodelService.updateSubmodelElement(idShortPathRootSME, newProperty);
+
+        Property updatedProperty = (Property) submodelService.getSubmodelElement(idShortPathRootSME);
```
Comment (mdanish98) [inline basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java:378]: Please rename updatedProperty to actualUpdatedProperty

**Review by mdanish98 — CHANGES_REQUESTED** [review]

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java (hunk 3 above)
Comment (ahoimariew) [reply basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java]: done

File: basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java [hunk 4]
```diff
@@ -230,14 +230,15 @@ public void updateNonFileSME() throws FileNotFoundException, IOException, ParseE
 	}
 
 	@Test
-	public void updateNonNestedSME() throws FileNotFoundException, IOException, ParseException {
-	        
+	public void updateNonNestedSME() throws FileNotFoundException, IOException, ParseException {        
 	    String submodelJSON = getJSONValueAsString("SingleSubmodelNew.json");
+
```
Comment (mdanish98) [inline basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java:235]: Please refactor

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java [hunk 5]
```diff
@@ -366,17 +366,20 @@ public void updateNonFileSME() {
...
         Property property = createDummyProperty("test123");
 
         submodelService.createSubmodelElement(property);
 
         String idShortPathRootSME = "test123";
 
-        Property newProperty = SubmodelServiceHelper.createDummyProperty("test123", "arbitraryValue", DataTypeDefXsd.STRING);
-        submodelService.updateSubmodelElement(idShortPathRootSME, newProperty);
+        Property expectedUpdatedProperty = SubmodelServiceHelper.createDummyProperty("test123", "arbitraryValue", DataTypeDefXsd.STRING);
+
```
Comment (mdanish98) [inline basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java:377]: No need for this space

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java [hunk 6]
```diff
@@ -366,17 +366,20 @@ public void updateNonFileSME() {
     public void updateNonNestedSME() {
         Submodel technicalSubmodel = DummySubmodelFactory.createTechnicalDataSubmodel();
         SubmodelService submodelService = getSubmodelService(technicalSubmodel);
+
         Property property = createDummyProperty("test123");
```
Comment (mdanish98) [inline basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java:371]: No need for this line

**Review by mdanish98 — CHANGES_REQUESTED** [review]

File: basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java (hunk 4 above)
Comment (ahoimariew) [reply basyx.submodelrepository/basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositorySubmodelHTTPTestSuite.java:235]: done

File: basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java (hunk 6 above)
Comment (ahoimariew) [reply basyx.submodelservice/basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelServiceSuite.java:371]: deleted

**Review by mdanish98 — APPROVED** [review]
Approved

**Pull request merged**